Ligand-based virtual screening

agitter opened the pull request:
This is an initial draft of the "Ligand-Based Prediction of Bioactivity" sub-section for the Treat section.  There are many remaining TODOs, but the main narrative is in place.  I attempted to capture the major feedback from #174.

@cgreene or @gwaygenomics can one of you please review this?

@rbharath I'm also interested in your opinion if you have time to read the first draft of this section, even if it is after we merge the text.  Many of the papers here are from you or the Pande lab.  I want to avoid over-hyping what is currently possible but also don't want to overcorrect and come across as pessimistic or dismissive of what has been accomplished recently.

For my own bookkeeping, this draft references issues #45, #52, #53, #54, #55, #57, #72, #75, #104, #141, #148, #173, #181, #228, #229, #232, #270.

rbharath commented:
@agitter I read through the draft and it looks pretty reasonable to me. The historical overview does a good job of capturing the early excitement about Merck's results, with the growing realization that the amount of "relatedness" between various tasks is very important for multitask boosts. I agree that the current excitement comes from the possibilities of learnable feature representations for molecules.

Engaging in armchair prophesying, I'd wager that we're entering a stage where education becomes more important: deep ligand-based networks are already quite useful tools in the hands of experts, but better tooling and incorporation into curricula will be necessary for them to get a broader reach.

I'd definitely recommend adding a section on deep-learning approaches to structural drug discovery. This field is certainly new, but there's been a burst of interesting work recently, from Atomwise's Atom-Net, 3D convolutions from David Koes' group, and atomic convolutions from our group.

cgreene commented on sections/05_treat.md:
```diff
@@ -162,45 +162,193 @@ feature selection and construction.*
+reviewing more existing literature on the topic` `TODO: check which other
+existing reviews should be referenced`
+
+We primarily focus on ligand-based approaches that train on chemicals' features
```
What do you mean by chemicals' features? Hydrophobicity, etc?

agitter replied:
Added a sentence to expand on chemicals' features

agitter replied:
@dhimmel do we have a way to cite a book?  I'd like to cite http://www.wiley.com/WileyCDA/WileyTitle/productCd-3527318526.html with ISBN 978-3-527-31852-0

dhimmel replied:
This book has a DOI: https://doi.org/10.1002/9783527628766

More generally citing books by ISBN would be nice... but I'd need to find an API that supports querying ISBNs and easily creating a CSL or bibtex entry. So not looking to do this work unless there is a compelling need.

agitter replied:
Perfect.  I searched the product page for a DOI and gave up too quickly.  I agree we don't need general ISBN support.

dhimmel replied:
Yeah I always check search.crossref.org (currently slow) since there can be a registered DOI that the publisher never actually displays.

cgreene commented on sections/05_treat.md:
```diff
@@ -162,45 +162,193 @@ feature selection and construction.*
-- Future output part 3 (most speculative), what would successful generative
-networks mean for the HTS field?
+In the biomedical domain, high-throughput chemical screening aims to improve
+therapeutic options over a long term horizon [@tag:PerezSianes2016_screening]
```
Not sure you need long term horizon here, unless you're going to contrast with other approaches that are shorter term (maybe categorizing patients for specific treatments is short term, while finding new molecules is long term)?

agitter replied:
Yes, that was the idea.  I was thinking ahead to the Treat intro where I'll distinguish between those two general areas.  Once I write the intro, I'll revisit how this sub-section's intro fits if needed.

cgreene commented on sections/05_treat.md:
```diff
@@ -162,45 +162,193 @@ feature selection and construction.*
+protein-protein interaction, or broader cellular phenotype.  This screening
+process can serve as the first step in the long drug discovery pipeline, where
+novel chemicals are pursued for their ability to inhibit or enhance
+disease-relevant biological mechanisms.  The appeal of machine learning in this
```
Do you need multiple hits though, or is one hit enough? Is the goal to identify a bunch of hits and then follow up with actual screens on only those hits? Does this help to give the most lead compound options in case one has to be abandoned due to off target activity?

cgreene commented on sections/05_treat.md:
```diff
@@ -162,45 +162,193 @@ feature selection and construction.*
+without exhaustively screening libraries of hundreds of thousands or millions of
+chemicals.  This task has been treated as a classification, regression, and
+ranking problem.  In reality, it does not fit neatly into any of those
+categories.  An ideal algorithm will rank a sufficient number of active
```
Is sufficient target dependent, or does it have a general meaning in this context (i.e. 10 lead compounds should be enough for anybody).

agitter replied:
I don't actually know.  I added a TODO and will see what I can find in the literature.

cgreene commented on sections/05_treat.md:
```diff
@@ -162,45 +162,193 @@ feature selection and construction.*
+Toxicology in the 21st Century (Tox21) Data Challenge, it did not dominate
+alternative methods as thoroughly as in other domains.  DeepTox was the top
+performer on only 9 of 15 targets, and in some cases there was little separation
+between DeepTox and the second best method on those 9 tasks.  A reliance on AUC
```
Was the second best method always the same, or were there many different second bests? How did it do on the 6/15 where it wasn't best? Did it crash and burn or do reasonably ok. These are not things you need to address - but if they lead to an interesting aside it may be worth including.

agitter replied:
It was 2nd best in 4 of those 6, 3rd in another (but very close to 2nd), and 5th in the worst case.  I tried to look into the 2nd best overall method but only found https://ncats.nih.gov/news/releases/2015/tox21-challenge-2014-winners and https://github.com/amaziz/Tox21-Challenge-Publication so far, which don't describe the algorithm.  I rephrased this sentence slightly.

cgreene commented on sections/05_treat.md:
```diff
@@ -162,45 +162,193 @@ feature selection and construction.*
+consequence of the degree to which tasks are related [@tag:Kearnes2016_admet].
+Some of the ADMET datasets showed far superior performance in multitask models
+of only the 22 ADMET tasks relative to more expansive multitask networks that
+included over 500 less-similar tasks.  `TODO: also has a good discussion of
```
We should cover this TODO in the discussion and could perhaps bring it back? We have some things like this in categorize too that touch on evaluation, and I know that came up in TF binding as well.

agitter replied:
Sounds good.  I'm leaving the TODO here for now to simplify the merge but can write about this in Discussion.

cgreene commented on sections/05_treat.md:
```diff
@@ -162,45 +162,193 @@ feature selection and construction.*
+between random forests and a deep neural were negligible, especially in the
+classification setting.  Overfitting is still a problem in larger chemical
+screening datasets with tens or hundreds of thousands of compounds because the
+number of active compounds can be very small, on the order of 0.1% or 1% of all
```
Is this the number of active compounds for `this` target or for `some` target?

agitter replied:
Added "for a typical target"

cgreene commented on sections/05_treat.md:
```diff
@@ -162,45 +162,193 @@ feature selection and construction.*
+approaches relied on preprocessing the graph into a feature vector, such as a
+fixed-width bit vector fingerprint [@tag:Rogers2010_fingerprints] in which each
+bit represents the presence of absence of a particular chemical substructure in
+the molecular graph. `TODO: this is a massive oversimplification because there
```
Maybe mention that this is an oversimplification and give a reference for more detail? I think that if I was getting into the domain I'd want to know this, but as a non-domain reader I'd probably scan by it.

agitter replied:
Rogers2010 is the reference.  I adapted the text.

cgreene commented on sections/05_treat.md:
```diff
@@ -162,45 +162,193 @@ feature selection and construction.*
+graphs have eliminated the need to enumerate artificial directed graphs,
+learning feature vectors for atoms that are a function of the properties of
+neighboring atoms and local regions on the molecular graph
+[@tag:Kearnes2016_graph_conv @tag:AltaeTran2016_one_shot]. `TODO: add more
```
Could refer back to the paper - was it this one ( https://arxiv.org/abs/1506.05163 ) from lecun's group? If I'm recalling my papers correctly, that one had the nuts and bolts about what was required for convolution over graphs.

agitter replied:
These two references each do their own type of graph convolution that are inspired by but different from that LeCun paper.  I'm going to elect to leave the details out because I don't think they influence the main point I want to make about learning feature representations.

cgreene commented on sections/05_treat.md:
```diff
@@ -162,45 +162,193 @@ feature selection and construction.*
+graph-based neural networks.  The simplified molecular-input line-entry system
+(SMILES) is a standard way to transform a chemical into string-based
+representation.  A SMILES-to-SMILES autoencoder learns a continuous latent
+feature space for chemicals [@tag:Gomezb2016_automatic].  In this learned
```
This connects to @brettbj's paper for EHRs discussed in categorize as well. Maybe worth connecting the dots on this to show consistent approaches across domains.

agitter replied:
TODO for now, I'll go back and read the EHR discussion

cgreene commented on sections/05_treat.md:
```diff
@@ -162,45 +162,193 @@ feature selection and construction.*
+compound and _one_ inactive compound, the model is able generalize reasonably
+well because it has learned an informative embedding function from the related
+tasks. Random forests, which cannot take advantage of the related training
+tasks, trained in the same setting achieve near random performance.  Despite the
```
achieve sounds very positive here - maybe `do not best random performance` or something that makes it clear that near random is not good.

agitter replied:
Fixed

cgreene commented on sections/05_treat.md:
```diff
@@ -162,45 +162,193 @@ feature selection and construction.*
+learned from the Tox21 assays have little utility for a drug adverse reaction
+dataset.
+
+In the long term, these novel, learned chemical feature representations may
```
Is this a different long term than what we started with? This one seems to be different to me. The first one is that taking a new molecule from identification -> drug is lengthy. This one seems more to be getting at fundamental biochemistry/pharmacology.

agitter replied:
Good point.  The `long term` here indicates that novel featurizations have not yet proved themselves in practical applications, but have the potential to do so with continued research.  I dropped `long term` here and the rest of the paragraph conveys the idea.

cgreene commented on sections/05_treat.md:
```diff
@@ -162,45 +162,193 @@ feature selection and construction.*
+network using standard fingerprints and slightly better than the neural
+fingerprints [@tag:Duvenaud2015_graph_conv] on the drug efficacy task but were
+slightly worse than the influence-relevance voter method on an HIV dataset.
+[@tag:Swamidass2009_irv].  `TODO: there are also problems with some papers using
```
Agree w/ further for discussion.

agitter replied:
Leaving the TODO here until we write the evaluation section of Discussion

cgreene commented on sections/05_treat.md:
```diff
@@ -162,45 +162,193 @@ feature selection and construction.*
+We remain optimistic for the potential of deep learning and specifically
+representation learning in this domain and propose that rigorous benchmarking on
+broad and diverse prediction tasks will be as important as novel neural network
+architectures to advance the state of the art _and convincingly demonstrate
```
Why is this italicized?

agitter replied:
I was trying to emphasize the discussion in #174 but it isn't needed.

cgreene commented on sections/05_treat.md:
```diff
@@ -162,45 +162,193 @@ feature selection and construction.*
+
+We remain optimistic for the potential of deep learning and specifically
+representation learning in this domain and propose that rigorous benchmarking on
+broad and diverse prediction tasks will be as important as novel neural network
```
Is it going to be about new architectures, more (public) data, or both? Trying to see how this fits w/ our broader themes.

agitter replied:
I'm hedging because I'm not sure.  Likely both.  I feel more confident that benchmarking, code sharing, and community platforms are going to be important.

cgreene commented on sections/05_treat.md:
```diff
@@ -162,45 +162,193 @@ feature selection and construction.*
+groups have already contributed additional datasets and prediction algorithms,
+and adoption of common benchmarking evaluation metrics, datasets, and baseline
+algorithms has the potential to establish the practical utility of deep learning
+in chemical bioactivity prediction and lower the barrier to entry for machine
```
Great! Can we provide a pointer to these resources? Or are they included in the above. If included, then maybe we should make that clearer by describing it as becoming a community project?

agitter replied:
Added `to DeepChem` to clarify it is indeed the above resource.  It's not quite a fully open community resource in that I think the main repository owners are mostly (entirely?) from one lab, but I did want to highlight community engagement and the eagerness to accept contributions from others.

agitter commented:
@rbharath Thank you very much for the feedback.  Structural drug discovery is definitely in my plans for the final version, and I believe the three papers you mentioned are the same ones I have in one of my TODOs.

I'm intrigued by your comments about education.  Are you referring to education about ML and available tools for (bio)chemists as opposed to chemistry education for ML practitioners?

agitter commented:
Thanks for the helpful review @cgreene.  I'd like to address some of these comments before merging and add TODOs for others.  It will have to wait a few days because I ignored some fairly important tasks this weekend to get this initial draft finished.

After this is merged, we'll have two major sub-sections in Treat.  I don't anticipate other contributions, so I'll work on wrapping up a full draft of the Treat section next weekend.

rbharath commented:
@agitter Yes, exactly. I think it will be much easier to educate (bio)chemists about ML than to educate ML practitioners about biology/chemistry. There's a large amount of good ML material and courses online already, and many universities are setting up strong ML classes in house. A student with (bio)chem background and basic ML knowledge can pick up deep-learning methods for drug discovery without too much trouble.

mrwns commented on sections/05_treat.md:
```diff
@@ -162,45 +162,193 @@ feature selection and construction.*
+Advances in chemical representation learning have not been limited to
+graph-based neural networks.  The simplified molecular-input line-entry system
+(SMILES) is a standard way to transform a chemical into string-based
+representation.  A SMILES-to-SMILES autoencoder learns a continuous latent
```
this earlier paper also describes how to perform ligand classification based on SMILES https://arxiv.org/abs/1602.06289
Even though it's a fascinating paper, I don't see the molecular autoencoder fitting here - its intent is to have a generative model for molecules to perform (neural) de-novo design, which would open another section, as there are also several other papers on the topic (GANs, RNNs...).

agitter replied:
Good point @mrwns.  I included the autoencoder here with the supervised methods because I want to focus on learned feature representations.  But that is an awkward fit.  I have a TODO to add GANs and other methods and will note that I need to revisit the autoencoder's placement as well.

agitter commented:
@cgreene I either addressed your comments in text or added new TODOs to address them during my weekened revisions.  I think we can merge once the DOI queries resolve.  Agreed?

cgreene commented:
@agitter : I agree. Looks like there are going to be some merge issues. Good luck!

agitter commented:
You're right, I think #334 introduced conflicts.  I'll clean it up before I merge this.
